aidial_adapter_bedrock/deployments.py

```diff
@@ -63,28 +63,11 @@ class ChatCompletionDeployment(str, Enum):
     COHERE_COMMAND_TEXT_V14 = "cohere.command-text-v14"
     COHERE_COMMAND_LIGHT_TEXT_V14 = "cohere.command-light-text-v14"
 
-    @property
-    def deployment_id(self) -> str:
-        """Deployment id under which the model is served by the adapter."""
-        return self.value
 
-    @property
-    def model_id(self) -> str:
-        """Id of the model in the Bedrock service."""
-
-        # Redirect Stability model without version to the earliest non-deprecated version (V1)
-        if self == ChatCompletionDeployment.STABILITY_STABLE_DIFFUSION_XL:
-            return (
-                ChatCompletionDeployment.STABILITY_STABLE_DIFFUSION_XL_V1.model_id
-            )
-
-        return self.value
-
-    @classmethod
-    def from_deployment_id(
-        cls, deployment_id: str
-    ) -> "ChatCompletionDeployment":
-        return cls(deployment_id)
+# Redirect Stability model without version to the earliest non-deprecated version (V1)
+CHAT_COMPLETION_REDIRECTS = {
```

Comment:
This stuff looks so similar to `COMPATIBILITY_MAPPING` logic
What do you think about including this case there too?

So instead of a dedicated variable for this "redirect" it would look something like this

```diff
class ChatCompletionDeployment(Enum):
    ...
-   STABILITY_STABLE_DIFFUSION_XL = "stability.stable-diffusion-xl"
    STABILITY_STABLE_DIFFUSION_XL_V1 = "stability.stable-diffusion-xl-v1"
```
```python
deployments = AdapterDeployments.create(
    compat_mapping=(
        # Redirect Stability model without version to the earliest non-deprecated version (V1)
        {"stability.stable-diffusion-xl": ChatCompletionDeployment.STABILITY_STABLE_DIFFUSION_XL_V1.value}
        | get_str_dict("COMPATIBILITY_MAPPING") 
    )
)
```

And in case, when user will want to redirect stability without version to latest version of XL, he'll just need to update env variable

Reply:
There is an important difference between redirects and compatibility modes.

Redirect is a simple model alias - you do a lookup in the table once and then carry the **destination name** everywhere.
In case of compatibility mode you need to carry around the **source name** because it will go to the upstream.

Mode | Adapter exposes endpoint for | Adapter uses API of | Upstream receives as a deployment ID | Result
-- | -- | -- | -- | --
redirect | my-fav-model | openai-model | openai-model | OK
compat | my-fav-model | openai-model | my-fav-model | FAIL

Reply:
Probably we simply need to deprecate `STABILITY_STABLE_DIFFUSION_XL` altogether. Let's leave it for another PR.

Reply:
https://github.com/epam/ai-dial-adapter-bedrock/issues/200